src/main/java/htsjdk/samtools/HtsgetBAMFileReader.java

```diff
@@ -0,0 +1,561 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.filter.FilteringSamIterator;
+import htsjdk.samtools.filter.SamRecordFilter;
+import htsjdk.samtools.util.*;
+import htsjdk.samtools.util.htsget.HtsgetClass;
+import htsjdk.samtools.util.htsget.HtsgetRequest;
+import htsjdk.samtools.util.zip.InflaterFactory;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.*;
+
+/**
+ * Class for reading and querying BAM files from an htsget source
+ */
+public class HtsgetBAMFileReader implements SamReader.PrimitiveSamReader {
```

Comment:
The documentation for ReaderImplementation mentions that it's specifically for file based readers as opposed to non file based readers, and that the class might need to be removed anyway, so I thought it would make more sense to extend directly from PrimitiveSamReader